src/finalizer/utils/polygon.ts

```diff
@@ -28,35 +47,45 @@ export async function getPosClient(mainnetSigner: Wallet) {
 }
 
 export async function getFinalizableTransactions(
+  logger: winston.Logger,
   tokensBridged: TokensBridged[],
   posClient: POSClient,
   hubPoolClient: HubPoolClient
 ) {
   const isCheckpointed = await Promise.all(
     tokensBridged.map((event) => posClient.exitUtil.isCheckPointed(event.transactionHash))
   );
-  const withdrawExitedOrIsNotCheckpointed = await Promise.all(
+  const exitStatus = await Promise.all(
     tokensBridged.map((event, i) => {
-      if (!isCheckpointed[i]) return new Promise((resolve) => resolve(true));
+      if (!isCheckpointed[i])
+        return new Promise((resolve) => resolve({ status: POLYGON_MESSAGE_STATUS.NOT_CHECKPOINTED }));
       const l1TokenCounterpart = hubPoolClient.getL1TokenCounterpartAtBlock(
-        "137",
+        CHAIN_ID.toString(),
         event.l2TokenAddress,
         hubPoolClient.latestBlockNumber
       );
-      return posClient.erc20(l1TokenCounterpart, true).isWithdrawExited(event.transactionHash);
+      return posClient.erc20(l1TokenCounterpart, true).isWithdrawExited(event.transactionHash)
+        ? new Promise((resolve) => resolve({ status: POLYGON_MESSAGE_STATUS.ALREADY_EXITED }))
+        : new Promise((resolve) => resolve({ status: POLYGON_MESSAGE_STATUS.CAN_EXIT }));
```

Comment:
I assume this is an async method. If so, I think this is probably what you want:
```suggestion
      return posClient.erc20(l1TokenCounterpart, true).isWithdrawExited(event.transactionHash).then((result) => result ?
        { status: POLYGON_MESSAGE_STATUS.ALREADY_EXITED } :
        { status: POLYGON_MESSAGE_STATUS.CAN_EXIT });
```